### DISCUSSION TITLE
Data older than 90 days from GET_V2_SETTLEMENT_REPORT_DATA_FLAT_FILE

### DISCUSSION THREAD

**Discussion by ASchmidtGit** [amzn/selling-partner-api-models#3582] [General]
Hey, we need the fees of orders from the last 4 Months. But the Data in GET_V2_SETTLEMENT_REPORT_DATA_FLAT_FILE seems to be limited to the last 90 days (allthough sellerboard can show way older data). Is there another report for this or am i missing something?

**Comment by moltar** [comment]
Finance API

**Reply by JoeChan-develop** [reply]
Hi, could you show us which specific API is ? Thanks

**Comment by JoeChan-develop** [comment]
Hi mate, I meet the same problem, have you find the answer yet ? We want to fetch the data more than 90days old(actually we need 2 years) ,but the SP-API doesn't work

**Comment by ritikacd** [comment]
Hi @ASchmidtGit and @JoeChan-develop, 

You can retrieve upto 180 days old finance data using [Finance APIs](https://developer-docs.amazon.com/sp-api/docs/finances-api-reference). For older than 180 days, it is not supported via SP-APIs. We suggest checking with the Seller Central support to inquire if this information is available. 

Thanks,
Ritika
Selling Partner API Services

**Reply by kevnhowe** [reply]
You can request and download every financial statement for all transactions going back years using the "Payments - All Statements" Seller Central page. We have used this page to download historical records going beyond 180 days. You must request the report for each settlement period, then download the report. It is available as both Flat File and XML.
We have created an app that processes financial settlement transactions via the "GET_V2_SETTLEMENT_REPORT_DATA_XML" API report. Given the result xml that comes as the payload of the API request is the same as the Seller Central report, you can easily create an alternative method that instead processes the xml version of the report you can download from the Seller Central page.
I cannot speak to how this would be done with the flat file, but the same logic should hold true.
